vignettes/howtogenefilter.Rmd

```diff
@@ -0,0 +1,190 @@
+the filter is *specific* or *non-specific*. Here, specific means that we
+are filtering with reference to sample metadata, for example, `type`.
+For example, if we want to select genes that are differentially
+expressed in the two groups defined by `type`, that is a specific
+filter. If on the other hand we want to select genes that are expressed
+in more than 5 samples, that is an example of a non--specific filter.
+
+First, let us see how to perform a non--specific filter. Suppose we want
+to select genes that have an expression measure above 200 in at least 5
+samples. To do that we use the function `kOverA`.
+
+There are three steps that must be performed.
+
+1.  Create function(s) implementing the filtering criteria.
+
+2.  Assemble it (them) into a (combined) filtering function.
+
+3.  Apply the filtering function to the expression matrix.
+
+```{r}
+f1 <- kOverA(5, 200)
+ffun <- filterfun(f1)
+wh1 <- genefilter(exprs(sample.ExpressionSet), ffun)
+sum(wh1)
+```
+
+Here `f1` is a function that implies our "expression measure above 200
+in at least 5 samples" criterion, the function `ffun` is the filtering
+function (which in this case consists of only one criterion), and we
+apply it using `genefilter`. There were genes that satisfied the
+criterion and passed the filter.
+As an example for a specific filter, let us select genes that are
+differentially expressed in the groups defined by `type`.
+
+```{r}
+f2 <- ttest(sample.ExpressionSet$type, p=0.1)
+ffun <- filterfun(f1)
+wh2 <- genefilter(exprs(sample.ExpressionSet), 
+filterfun(f2))
+sum(wh2)
+```
+Here, `ttest` is a function from the `genefilter`
+package which provides a suitable wrapper around \texttt{t.test} from
+package \textit{stats}. Now we see that there are  `r \Sexpr{sum(wh2)}`
+genes that satisfy the selection criterion. 
+
+Suppose that we want to combine the two filters. We want those genes
+for which at least 5 have an expression measure over 200 *and* which also are differentially
+expressed between the groups defined by `type+`.
+
+```{r}
+ffun_combined <- filterfun(f1, f2)
+wh3 <- genefilter(exprs(sample.ExpressionSet), ffun_combined)
+sum(wh3)
+```
+
+Now we see that there are only \Sexpr{sum(wh3)} genes  that satisfy both conditions.
+
+FIXME: need to replace this with something else
+Our last example is to select genes that are
+differentially expressed in at least one of the three groups defined
+by \verb+cov3+.
+To do that we use an Anova filter. This filter uses an analysis of
+variance appraoch (via the \verb+lm+) function to test the hypothesis
+that at least one of the three group means is different from the other
+two. The test is applied, then the $p$--value computed. We select
+%those genes that have a low $p$--value.
+
+
+Afilter <- Anova(eset$cov3)
+aff <- filterfun(Afilter)
+wh4 <- genefilter(exprs(eset), aff)
+sum(wh4)
+
+We see that there are 14 genes that pass this filter and that are
+candidates for further exploration.
+
```

Comment:
Remove lines 101-119. These were likely comments in the .Rnw document to the author that do not appear in the PDF.